[BREAKING] Remove `dwo` as a sensor and add it as a number entity

Triple-S opened the pull request:
Very simple change to make the dwo field writeable.

syssi commented:
Could you update the `README.md` too? (Move the `dwo` row to the number section)

Triple-S commented:
> Could you update the `README.md` too? (Move the `dwo` row to the number section)

Sure, it is done.

Triple-S commented:
I just realized that there was another flaw with the implementation: Initially I assumed that setting the Charging energy limit to 0 would remove the limit since it is displayed like this in the go-e app. However, I found out that internally there is a distinct difference between a limit of 0 and "null" (unset). Because I don't see any reason to set the limit to exactly 0 on purpose, I added an exception in number.py to report a 0 as "null" to the charger and thereby remove the limit. In Home Assistant this state is displayed as an empty field. One could add another exception to display this state as 0, but I think it is better to have a visibile difference between 0 and "null" in Home Assiatant.

Another change I implemented is increasing the maximum possible value of the Charging energy limit from 100kWh to 1000kWh since there are already cars around with batteries larger than 100kWh, so the maximum could have been too low for some cars.

syssi commented on custom_components/goecharger_mqtt/number.py:
```diff
@@ -43,10 +43,13 @@ def __init__(
         """Update the current value."""
-        if self.native_step == 1:
-            await mqtt.async_publish(self.hass, f"{self._topic}/set", int(value))
+        if self._topic.endswith("/dwo") and value == 0:
```
I don't like this special case. Could you extend the `GoEChargerSensorEntityDescription` call by a new property which can be enabled per definition? This would be generic and reusable for other entities.

Triple-S replied:
Yes, you are right, it is done. I guess you meant to add it to `GoEChargerNumberEntityDescription`.

syssi commented on custom_components/goecharger_mqtt/number.py:
```diff
@@ -43,13 +43,12 @@ def __init__(
     async def async_set_native_value(self, value: float) -> None:
         """Update the current value."""
-        if self._topic.endswith("/dwo") and value == 0:
+        if self.entity_description.treat_zero_as_null and value == 0:
```
Could you swap the conditions? :-) Thanks in advance!
